### PR TITLE
Expose publish options

### PR DESCRIPTION
**Pull request by elucidsoft**
This resolves PR #71

### CODE REVIEW

**Comment by elucidsoft** [comment]
Fixing build...

File: packages/rabbitmq/src/amqp/connection.ts [hunk 1]
```diff
@@ -77,7 +78,8 @@ export class AmqpConnection {
 
     this.publish(requestOptions.exchange, requestOptions.routingKey, payload, {
       replyTo: DIRECT_REPLY_QUEUE,
-      correlationId
+      correlationId,
+      ...publishOptions
```
Comment (WonderPanda) [inline packages/rabbitmq/src/amqp/connection.ts:82]: The signature for `publish` is:
```
public publish(
    exchange: string,
    routingKey: string,
    message: any,
    options?: amqplib.Options.Publish
  )
```
So it's expecting the options as a 4th arg.  Looks like you're spreading them into the 3rd (message) arg here so I don't think it'll interact properly with the underlying amqplib

**Review by WonderPanda — CHANGES_REQUESTED** [review]
This repo depends on using yarn to properly leverage yarn workspaces + lerna to support monorepo publishing. So there shouldn't be a package-lock.json included

File: packages/rabbitmq/src/amqp/connection.ts (hunk 1 above)
Comment (elucidsoft) [reply packages/rabbitmq/src/amqp/connection.ts:82]: let me double check I did this right...

Comment (WonderPanda) [reply packages/rabbitmq/src/amqp/connection.ts:82]: Whoops nevermind I misread this

Comment (elucidsoft) [reply packages/rabbitmq/src/amqp/connection.ts:82]: Yes, I wanted to keep the values that were specific to request but allow additional items to be added.

**Comment by elucidsoft** [comment]
I would like to abandon this...

**Pull request closed without merging**